trunk/SUAVE/Methods/Performance/electric_V_h_diagram.py

```diff
@@ -0,0 +1,175 @@
+                            display_plot = True,
+                            climb_rate_contours = [0.]
+                            ):
+
+        Calculates and optionally displays climb rate and contours thereof over
+        a specified airspeed and altitude range. Climb rate determination ref.
+        Raymer, "Aircraft Design: A Conceptual Approach"
+
+        Sources:
+        D. Raymer, "Aircraft Design: A Conceptual Approach"
+
+        Assumptions:
+
+        Assumes use of Battery Propeller Energy Network
+
+        Inputs:
+
+            vehicle                         SUAVE Vehicle Structure
+                .mass_properties
+                    .takeoff                                            [kg]
+            analyses                        SUAVE Analyses Structure
+                .atmosphere
+                    .planet
+                        .sea_level_gravity                              [m/s^2]
+            delta_isa                       ISA Temperature Offset      [deg. K/C]
+            grid_points                     Num. Test Points per Dim.   [Int]
+            altitude_ceiling                Maximum Test Altitude       [User Set]
+            max_speed                       Maximum Test Speed          [User Set]
+            test_omega                      Maximum Power Prop Speed    [User Set]
+            display_plot                    Flag for Plot Generation    [Boolean]
+            climb_rate_contours             Climb Rates to Display      [ft/min]
+
+        Outputs:
+
+            climb_rate                      Climb Rates at Test Points  [ft/min]
+    """
+
+    # Unpack Inputs
+
+    g       = analyses.atmosphere.planet.sea_level_gravity
+    W       = vehicle.mass_properties.takeoff * g
+    S       = vehicle.reference_area
+
+    # Single Point Mission for Drag Determination
+
+    def mini_mission(altitude, speed):
+
+        mission = SUAVE.Analyses.Mission.Sequential_Segments()
+        mission.tag = 'the_mission'
+
+        segment = SUAVE.Analyses.Mission.Segments.Single_Point.Set_Speed_Set_Altitude_No_Propulsion()
+        segment.tag = 'single_point'
+        segment.analyses.extend(analyses)
+        segment.altitude = altitude
+        segment.air_speed = speed
+
+        mission.append_segment(segment)
+
+        return mission
+
+    # Specify Altitude and Speed Sample Points
+
+    alt_range       = np.linspace(0., altitude_ceiling, num=grid_points, endpoint=True)
+    speed_range     = np.linspace(0., max_speed, num = grid_points, endpoint=False)
+
+    # Initialize Climb Rate Grid
+
+    climb_rate = np.zeros((grid_points, grid_points))
+
+    # Loop Through Altitude and Speed Gridpoints
+
+    for alt_idx in range(grid_points):
+
+        altitude    = alt_range[alt_idx]
+        atmo_data   = analyses.atmosphere.compute_values(altitude, delta_isa)
+        rho         = atmo_data.density
+        Vs          = np.sqrt(2*W/(rho*S*1.4)) # Determine Vehicle Stall Speed
```

Comment:
Is 1.4 a buffer for stall speed assumed by raymer? Is this something that he expects to be varied?

Reply:
This was a holdover, hardcoded CL_max. It's been changed to be user set now.